src/cast_core/test/CrudActorYamlTests.yaml

```diff
@@ -0,0 +1,185 @@
+    OutcomeData:
+      - {a: 2}
+
+  - Description: Inserts and replaces document in the database.
+    Operations:
+      - OperationName: bulkWrite
+        OperationCommand:
+          WriteOperations:
+            - WriteCommand: insertOne
+              Document: { a: 1 }
+            - WriteCommand: replaceOne
+              Filter: { a : 1 }
+              Replacement: { name: test }
+    OutcomeData:
+      - {name: test}
+
+  - Description: Inserts and updates with 'BypassDocumentValidation' true and 'Ordered' false.
+    Operations:
+      - OperationName: bulkWrite
+        OperationCommand:
+          WriteOperations:
+            - WriteCommand: insertOne
+              Document: { a: 1 }
+            - WriteCommand: updateOne
+              Filter: { a: 1 }
+              Update: { $set: { a: 5 } }
+          Options:
+            Ordered: false
+            BypassDocumentValidation: true
+    OutcomeData:
+      - {a: 5}
+    ExpectedEvents:
+      - {ordered: false, bypassDocumentValidation: true}
+
+  - Description: Inserts and updates many documents in the database.
+    Operations:
+      - OperationName: bulkWrite
+        OperationCommand:
+          WriteOperations:
+            - WriteCommand: insertOne
+              Document: { a: {^RandomInt: {min: 5, max: 15} } }
+            - WriteCommand: insertOne
+              Document: { a: {^RandomInt: {min: 5, max: 15} } }
+            - WriteCommand: insertOne
+              Document: { a: {^RandomInt: {min: 5, max: 15} } }
+            - WriteCommand: updateMany
+              Filter: { a: {$gte: 5} }
+              Update: { $set: { a: 1 } }
+    OutcomeCounts:
+      - Filter: {a: 1}
+        Count: 3
+
+  - Description: Inserts and deletes many documents in the database.
+    Operations:
+      - OperationName: bulkWrite
+        OperationCommand:
+          WriteOperations:
+            - WriteCommand: insertOne
+              Document: { a: {^RandomInt: {min: 5, max: 15} } }
+            - WriteCommand: insertOne
+              Document: { a: {^RandomInt: {min: 5, max: 15} } }
+            - WriteCommand: insertOne
+              Document: { a: {^RandomInt: {min: 5, max: 15} } }
+            - WriteCommand: deleteMany
+              Filter: { a: {$gte: 5} }
+    OutcomeCounts:
+      - Filter: {a: {$gte: 5}}
+        Count: 0
+
+  - Description: Insert randomly generated doc and update with write concern majority.
+    Operations:
+      - OperationName: bulkWrite
+        OperationCommand:
+          WriteOperations:
+            - WriteCommand: insertOne
+              Document: { a: {^RandomInt: {min: 2, max: 5} }}
+            - WriteCommand: updateOne
+              Filter: { a: { $gte: 2 } }
+              Update: { $set: { a: 8 } }
+          Options:
+            WriteConcern:
+              Level: majority
+              Timeout: 6000 milliseconds
+    OutcomeCounts:
+      - Filter: {a: 8}
+        Count: 1
+    ExpectedEvents:
```

Comment:
Should this be a list of all events or just that there exists an "event" in all events where "event" is in the list of  `ExpectedEvents`?

Reply:
bug in my own understanding - tests want to assert facts about _all_ callback events. Latest version has it.